repository/git.go

```diff
@@ -769,27 +993,50 @@ func (repo *GitRepo) PushNotesAndArchive(remote, notesRefPattern, archiveRefPatt
 	return nil
 }
 
+func (repo *GitRepo) getRefHashes(refPattern string) (map[string]string, error) {
+	if !strings.HasSuffix(refPattern, "/*") {
+		return nil, fmt.Errorf("unsupported ref pattern %q", refPattern)
+	}
+	refPrefix := strings.TrimSuffix(refPattern, "*")
+	showRef, err := repo.runGitCommand("show-ref")
+	if err != nil {
+		return nil, err
+	}
+	refsMap := make(map[string]string)
+	for _, line := range strings.Split(showRef, "\n") {
+		lineParts := strings.Split(line, " ")
+		if len(lineParts) != 2 {
+			return nil, fmt.Errorf("unexpected line in output of `git show-ref`: %q", line)
+		}
+		if strings.HasPrefix(lineParts[1], refPrefix) {
+			refsMap[lineParts[1]] = lineParts[0]
+		}
+	}
+	return refsMap, nil
+}
+
 func getRemoteNotesRef(remote, localNotesRef string) string {
 	relativeNotesRef := strings.TrimPrefix(localNotesRef, "refs/notes/")
-	return "refs/notes/" + remote + "/" + relativeNotesRef
+	return "refs/notes/remotes/" + remote + "/" + relativeNotesRef
+}
+
+func getLocalNotesRef(remote, remoteNotesRef string) string {
+	relativeNotesRef := strings.TrimPrefix(remoteNotesRef, "refs/notes/remotes/"+remote+"/")
```

Comment:
formatting seems a little strange on the string concats of the second param here.

Reply:
If you mean the lack of spaces between values, that is because it is a method argument. The formatting is automatic and performed by `gofmt`